drivers/hwtrig/hwtrig_stm32_timer.c

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2016 Linaro Limited.
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ * Copyright (c) 2021 Shlomi Vaknin.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_timer_trigger
+
+#include <errno.h>
+
+#include <soc.h>
+#include <stm32_ll_rcc.h>
+#include <stm32_ll_tim.h>
+#include <drivers/hwtrig.h>
+#include <device.h>
+#include <kernel.h>
+#include <init.h>
+
+#include <drivers/clock_control/stm32_clock_control.h>
+#include <pinmux/stm32/pinmux_stm32.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(stm32_timer_trigger, CONFIG_HWTRIG_LOG_LEVEL);
+
+/* L0 series MCUs only have 16-bit timers and don't have below macro defined */
+#ifndef IS_TIM_32B_COUNTER_INSTANCE
+#define IS_TIM_32B_COUNTER_INSTANCE(INSTANCE) (0)
+#endif
+
+/** HWTRIG data. */
+struct hwtrig_stm32_data {
+	/** Timer clock (Hz). */
+	uint32_t tim_clk;
+};
+
+/** HWTRIG configuration. */
+struct hwtrig_stm32_config {
+	/** Timer instance. */
+	TIM_TypeDef *timer;
+	/** Prescaler. */
+	uint32_t prescaler;
+	/** Clock configuration. */
+	struct stm32_pclken pclken;
+};
+
+/**
+ * Obtain timer clock speed.
+ *
+ * @param pclken  Timer clock control subsystem.
+ * @param tim_clk Where computed timer clock will be stored.
+ *
+ * @return 0 on success, error code otherwise.
+ */
+static int get_tim_clk(const struct stm32_pclken *pclken, uint32_t *tim_clk)
```

Comment:
Looks like a re-use of the PWM `get_tim_clk`, could be made common maybe?

Reply:
I agree. Maybe another file called `timer_stm32_common.c`?